druid/src/widget/switch.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A toggle switch widget.
+
+use crate::kurbo::{Circle, Point, Rect, RoundedRect, Shape, Size};
+use crate::piet::{
+    FontBuilder, LinearGradient, RenderContext, Text, TextLayout, TextLayoutBuilder, UnitPoint,
+};
+use crate::theme;
+use crate::widget::Align;
+use crate::{
+    BaseState, BoxConstraints, Env, Event, EventCtx, LayoutCtx, LocalizedString, PaintCtx,
+    UpdateCtx, Widget,
+};
+
+/// A switch that toggles a boolean.
+#[derive(Debug, Clone, Default)]
+pub struct Switch {
+    knob_pos: Point,
+    knob_hovered: bool,
+}
+
+impl Switch {
+    pub fn new() -> impl Widget<bool> {
+        Align::vertical(UnitPoint::CENTER, Self::default())
+    }
+
+    fn knob_hit_test(&self, knob_width: f64, mouse_pos: Point) -> bool {
+        let knob_circle = Circle::new(self.knob_pos, knob_width / 2.);
+        knob_circle.winding(mouse_pos) > 0
+    }
+
+    fn paint_label(
+        &mut self,
+        paint_ctx: &mut PaintCtx,
+        base_state: &BaseState,
+        data: &bool,
+        env: &Env,
+        switch_width: f64,
+        switch_padding: f64,
+    ) {
+        let font_name = env.get(theme::FONT_NAME);
+        let font_size = env.get(theme::TEXT_SIZE_NORMAL);
+        let switch_height = env.get(theme::BORDERED_WIDGET_HEIGHT);
+
+        let label: LocalizedString<&str> = if *data {
```

Comment:
okay, I appreciate the effort but this won't really work. I really need to do a write up about how localization is supposed to function; in the meantime I think you should go back to just using a normal string and add a `TODO` that this should be localized at some point in the future. :)

Reply:
Okay, done :)